tests/python_functional/src/syslog_ng_config/renderer.py

```diff
@@ -20,6 +20,129 @@
 # COPYING for details.
 #
 #############################################################################
+from src.syslog_ng_config.statements.filters.filter import Filter
+
+
+def render_version(version):
+    return "@version: {}\n".format(version)
+
+
+def render_includes(includes):
+    include_lines = ['@include "{}"'.format(include) for include in includes]
+    return "\n".join(include_lines)
+
+
+def render_global_options(global_options):
+    globals_options_header = "options {\n"
+    globals_options_footer = "};\n"
+
+    config_snippet = globals_options_header
+    for option_name, option_value in global_options.items():
+        config_snippet += "    {}({});\n".format(option_name, option_value)
+    config_snippet += globals_options_footer
+
+    return config_snippet
+
+
+def render_positional_options(positional_parameters):
+    config_snippet = ""
+    for parameter in positional_parameters:
+        config_snippet += "        {}\n".format(str(parameter))
+    return config_snippet
+
+
+def render_dict(name, options):
+
+    config_snippet = ""
+    config_snippet += "        {}(\n".format(name)
+    config_snippet += render_driver_options(options)
+    config_snippet += "        )\n"
+
+    return config_snippet
+
+
+def render_list(name, options):
```

Comment:
If possible, could we change the function name? Maybe some config related name would be better.

Reply:
We discussed this with Micek, and for now we leave the name as it is. In the long run, we can eliminate its need. FYI: @norberttak 

The `render_list` functionality was introduced for  `snmp`, and `snmp` is it's only user. In case of `snmp`, the usual python dictionary cannot work, because the same keys may appear in the driver, but a key can appear only once in a dictionary.

I am thinking of introducing a new data structure: let's call it `MultiDict`, that can store the same key multiple times. If it has the same `items()` interface used for rendering, and returns the multi-occurring keys too, the original render function can work without introducing the `render_list`. That would be a nice dependency reversal.